### PR TITLE
Consolidate Service Bus samples

### PR DESCRIPTION
**Pull request by conniey**
# Description

* Consolidates the service bus samples into fewer classes
* Removes duplicated samples
* Removes the use of connection string in samples.

# All SDK Contribution checklist:
- [x] **The pull request does not introduce [breaking changes]**
- [x] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md).**

## [General Guidelines and Best Practices](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#developer-guide)
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#building-and-unit-testing)
- [x] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: sdk/servicebus/azure-messaging-servicebus/README.md [hunk 1]
```diff
@@ -185,19 +164,37 @@ a topic.
...
+    // The batch is full. Send the current batch and create a new one.
+    sender.sendMessages(batch);
+
+    batch = sender.createMessageBatch();
+
+    batch.tryAddMessage(message);
+
+    // Add the message we couldn't before.
+    if (!batch.tryAddMessage(message)) {
+        throw new IllegalArgumentException("Message is too large for an empty batch.");
```
Comment (lmolkova) [inline sdk/servicebus/azure-messaging-servicebus/README.md:188]: something is off with this sample:
- on line 184 we add a message
- now we add another one and tell that batch is empty

**Review by lmolkova — APPROVED** [review]

File: sdk/servicebus/azure-messaging-servicebus/README.md [hunk 2]
```diff
@@ -79,33 +79,6 @@ add the direct dependency to your project as follows.
...
-    .connectionString("<< CONNECTION STRING FOR THE SERVICE BUS NAMESPACE >>")
-    .receiver()
-    .topicName("<< TOPIC NAME >>")
-    .subscriptionName("<< SUBSCRIPTION NAME >>")
-    .buildAsyncClient();
-```
-
 #### Create a Service Bus client using Microsoft Identity platform (formerly Azure Active Directory)
 
 Azure SDK for Java supports the Azure Identity package, making it simple to get credentials from the Microsoft identity
```
Comment (joshfree) [inline sdk/servicebus/azure-messaging-servicebus/README.md:84]: nit: package => client library

Comment (joshfree) [reply sdk/servicebus/azure-messaging-servicebus/README.md:84]: nit: the Microsoft identity => Azure AD

**Review by joshfree — APPROVED** [review]

**Review by anuchandy — APPROVED** [review]

File: sdk/servicebus/azure-messaging-servicebus/README.md (hunk 1 above)
Comment (conniey) [reply sdk/servicebus/azure-messaging-servicebus/README.md:188]: Good catch!

**Pull request merged**